### PR TITLE
[GAL-5440] add Suggested Profiles in curated Feed web

### PR DESCRIPTION
**Pull request by Rohan-cp**
### Summary of Changes

adds a new item (the Suggested Profiles section) in the `Curated` Feed after the 8th feed list item.

### Demo or Before/After Pics

- If this is a new feature, include screenshots or recordings of the feature in action.
- If this PR makes visual changes, include before and after screenshots.

| Design                                     | Demo with changes                                      |
| ------------------------------------------ | ------------------------------------------ |
| Screenshot 2024-04-17 at 12 07 17 | Screenshot 2024-04-17 at 12 07 29|

### Edge Cases
Moweb:
Screenshot 2024-04-17 at 13 06 36

### Testing Steps

Provide steps on how the reviewer can test the changes.

### Checklist

Please make sure to review and check all of the following:

- [x] I've tested the changes and all tests pass.
- [x] (if web) I've tested the changes on various desktop screen sizes to ensure responsiveness.
- [ ] (if mobile) I've tested the changes on both light and dark modes.

### CODE REVIEW

File: apps/web/src/components/Feed/CuratedFeed.tsx [hunk 1]
```diff
@@ -89,6 +89,7 @@ export default function CuratedFeed({ queryRef }: Props) {
       loadNextPage={loadNextPage}
       hasNext={hasPrevious}
       feedMode={'WORLDWIDE'}
+      showSuggestedProfiles={true}
```
Comment (Robinnnnn) [inline apps/web/src/components/Feed/CuratedFeed.tsx:92]: nit: can just pass in `showSuggestedProfiles` without setting to true

```js
  feedMode={'WORLDWIDE'}
  showSuggestedProfiles
  ...
```

File: apps/web/src/components/Feed/FeedList.tsx [hunk 2]
```diff
@@ -59,11 +64,36 @@ export default function FeedList({
     feedEventRefs
   );
 
+  const isMobileOrMobileLargeWindowWidth = useIsMobileOrMobileLargeWindowWidth();
+  const suggestedProfileSectionHeight = useMemo(
+    () => (isMobileOrMobileLargeWindowWidth ? 320 : 360),
+    [isMobileOrMobileLargeWindowWidth]
+  );
```
Comment (Robinnnnn) [inline apps/web/src/components/Feed/FeedList.tsx:71]: nit: don't need to memoize primitive types; memoization mostly meant for complex types like computed values, arrays, objects. 

can just do:
```js
const suggestedProfileSectionHeight = isMobileOrMobileLargeWindowWidth ? 320 : 360
```

File: apps/web/src/components/Feed/FeedList.tsx [hunk 3]
```diff
@@ -59,11 +64,36 @@ export default function FeedList({
...
+
+  // insert suggested profiles in between posts if showSuggestedProfiles is true
+  const finalFeedData = useMemo(() => {
+    const suggestedProfileSectionIdx = 8;
+    if (showSuggestedProfiles && feedData?.length >= suggestedProfileSectionIdx) {
+      const suggestedProfileSectionData = {
+        __typename: 'SuggestedProfileSection',
+        dbid: '12345',
```
Comment (Robinnnnn) [inline apps/web/src/components/Feed/FeedList.tsx:79]: let's use `uuid()`, we use it in a couple other places in the repo

File: apps/web/src/components/Feed/FeedList.tsx [hunk 4]
```diff
@@ -196,10 +245,31 @@ export default function FeedList({
...
       virtualizedListRef.current?.recomputeRowHeights();
     },
-    [feedData, measurerCache]
+    [finalFeedData, measurerCache]
   );
 
-  const rowCount = hasNext ? feedData.length + 1 : feedData.length;
+  const rowCount = hasNext ? finalFeedData.length + 1 : finalFeedData.length;
+
+  const rowHeight = useCallback(
```
Comment (Robinnnnn) [inline apps/web/src/components/Feed/FeedList.tsx:253]: nice work figuring this out!

File: apps/web/src/components/Feed/FeedSuggestedProfileSection.tsx [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+  const handleProfileClick = useCallback(
+    (username: string) => {
+      const path = {
+        pathname: '/[username]',
+        query: { username: username },
+      } as Route;
+
+      if (!path) {
+        return;
+      }
```
Comment (Robinnnnn) [inline apps/web/src/components/Feed/FeedSuggestedProfileSection.tsx:66]: don't think we need the check on L64-66 if `path` is clearly defined on L59?

File: apps/web/src/components/Feed/FeedSuggestedProfileSection.tsx [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+  const query = useFragment(
+    graphql`
+      fragment FeedSuggestedProfileSectionWithBoundaryFragment on Query {
+        ...FeedSuggestedProfileSectionQueryFragment
+      }
+    `,
+    queryRef
+  );
+  return (
+    <ReportingErrorBoundary fallback={<></>}>
```
Comment (Robinnnnn) [inline apps/web/src/components/Feed/FeedSuggestedProfileSection.tsx:191]: should we have skeleton states?

File: apps/web/src/components/Feed/CuratedFeed.tsx (hunk 1 above)
Comment (Rohan-cp) [reply apps/web/src/components/Feed/CuratedFeed.tsx]: made the change!

File: apps/web/src/components/Feed/FeedSuggestedProfileSection.tsx (hunk 6 above)
Comment (Rohan-cp) [reply apps/web/src/components/Feed/FeedSuggestedProfileSection.tsx:188]: right I was just following the structure for the other feed items and they didn't have a `ReportingErrorBoundary` or `Suspense` fallback (attached sc of one). So do you think we should add the skeleton state here for this section?

Screenshot 2024-04-17 at 14 49 22

**Review by Robinnnnn — APPROVED** [review]

**Pull request merged**